tests/visualization_tests/test_contour.py

```diff
@@ -79,252 +108,443 @@ def fail_objective(_: Trial) -> float:
+    yaxis = info.sub_plot_infos[0][0].yaxis
+    assert xaxis.name == "param_b"
+    assert yaxis.name == "param_d"
+    assert xaxis.range == (-0.05, 1.05)
+    assert yaxis.range == (1.9, 4.1)
+    assert not xaxis.is_log
+    assert not yaxis.is_log
+    assert not xaxis.is_cat
+    assert not yaxis.is_cat
+    assert xaxis.indices == [-0.05, 0.0, 1.0, 1.05]
+    assert yaxis.indices == [1.9, 2.0, 4.0, 4.1]
+    assert xaxis.values == [0.0, 1.0]
+    assert yaxis.values == [4.0, 2.0]
+    z_values = info.sub_plot_infos[0][0].z_values
+    assert z_values == {(1, 2): 2.0, (2, 1): 1.0}
+
+
+@pytest.mark.parametrize("objective", (0, 1))
+@pytest.mark.parametrize("value", (float("inf"), -float("inf")))
+def test_test_get_contour_info_nonfinite_multiobjective(objective: int, value: float) -> None:
+
+    study = prepare_study_with_trials(with_c_d=True, n_objectives=2, value_for_first_trial=value)
+    info = _get_contour_info(
+        study, params=["param_b", "param_d"], target=lambda t: t.values[objective]
+    )
+    assert info.sorted_params == ["param_b", "param_d"]
+    assert np.shape(np.asarray(info.sub_plot_infos)) == (1, 1, 3)
+    xaxis = info.sub_plot_infos[0][0].xaxis
+    yaxis = info.sub_plot_infos[0][0].yaxis
+    assert xaxis.name == "param_b"
+    assert yaxis.name == "param_d"
+    assert xaxis.range == (-0.05, 1.05)
+    assert yaxis.range == (1.9, 4.1)
+    assert not xaxis.is_log
+    assert not yaxis.is_log
+    assert not xaxis.is_cat
+    assert not yaxis.is_cat
+    assert xaxis.indices == [-0.05, 0.0, 1.0, 1.05]
+    assert yaxis.indices == [1.9, 2.0, 4.0, 4.1]
+    assert xaxis.values == [0.0, 1.0]
+    assert yaxis.values == [4.0, 2.0]
+    z_values = info.sub_plot_infos[0][0].z_values
+    assert z_values == {(1, 2): 2.0, (2, 1): 1.0}
 
 
 @pytest.mark.parametrize("direction", ["minimize", "maximize"])
 def test_color_map(direction: str) -> None:
     study = prepare_study_with_trials(with_c_d=False, direction=direction)
 
     # `target` is `None`.
-    contour = plot_contour(study).data[0]
+    contour = plotly_plot_contour(study).data[0]
     assert COLOR_SCALE == [v[1] for v in contour["colorscale"]]
     if direction == "minimize":
         assert contour["reversescale"]
     else:
         assert not contour["reversescale"]
 
     # When `target` is not `None`, `reversescale` is always `True`.
-    contour = plot_contour(study, target=lambda t: t.number).data[0]
+    contour = plotly_plot_contour(study, target=lambda t: t.number).data[0]
     assert COLOR_SCALE == [v[1] for v in contour["colorscale"]]
     assert contour["reversescale"]
 
     # Multi-objective optimization.
     study = prepare_study_with_trials(with_c_d=False, n_objectives=2, direction=direction)
-    contour = plot_contour(study, target=lambda t: t.number).data[0]
+    contour = plotly_plot_contour(study, target=lambda t: t.number).data[0]
     assert COLOR_SCALE == [v[1] for v in contour["colorscale"]]
     assert contour["reversescale"]
```

Comment:
Should this also be tested about `_get_contour_info`, instead of `plotly_plot_contour`?

Reply:
Hmm... This is not a strong opinion, but the current one is simple enough, so we can leave it as it is. The main scope of this test is the color map of the generated figure, so both `plotly_plot_contour` and `_get_contour_info` are fine, I think.